### PR TITLE
New auth connect #4

### PR DESCRIPTION
**Pull request by scsirdx**

### CODE REVIEW

File: app/api.js [hunk 1]
```diff
@@ -1,111 +1,38 @@
...
-    fetch(`${localApiUrl}oauth/token`, {
-      method: 'POST',
-      headers: prepareHeaders(),
-      body: JSON.stringify({ grant_type: 'refresh_token', refresh_token: refreshToken }),
-    })
-    .then(parseResponse)
-    .then(data => resolve(normalizeToken(data)))
-    .catch(errors => reject(errors));
-  });
+  if (token) headers.Authorization = `Bearer ${token}`;
```
Comment (mkristian) [inline app/api.js:16]: not sure if we need to prefix with `Bearer` here, but server might just except it like this.

File: app/sagas.js [hunk 2]
```diff
@@ -1,80 +1,53 @@
...
-      password = null;
-
-      yield race({
-        delay: call(delay, newToken.expiresIn - 600 * 1000),
-        refresh: take(constants.REFRESH_TOKEN),
-      });
+export function* ping({ apiUrl, apiPath, token }) {
+  while (true) {
+    try {
+      yield delay(1000 * 60);
```
Comment (mkristian) [inline app/sagas.js:12]: let's see if we can keep this request out of the KEEP-THIS-SESSION-ALIVE trigger :)
that is a ping each minute - correct ?

**Review by mkristian — APPROVED** [review]
Not sure whether each request can update the stored token - could not find the right piece of code for this. token get parsed with each response but the store bits I missed.

otherwise looks good to me

File: app/sagas.js (hunk 2 above)
Comment (scsirdx) [reply app/sagas.js:12]: Yes. Later ping will be configurable in UI setting. Something like "log out after XX minutes of inactivity"

**Pull request merged**